src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -132,115 +230,867 @@ static DamageTable D_8092C140 = {
+                ActorCutscene_Stop(this->actor.cutscene);
+                Actor_MarkForDeath(&this->actor);
+            }
+        }
+        return;
+    }
+    if (Animation_OnFrame(&this->skelAnime, 23.0f)) {
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_WALK);
+    }
+}
+
+void EnIk_SetupFrozen(EnIk* this) {
+    this->invincibilityFrames = 0;
+    this->actionFunc = EnIk_Frozen;
+    this->actor.speedXZ = 0.0f;
+}
+
+void EnIk_Frozen(EnIk* this, GlobalContext* globalCtx) {
+    if (this->counter) {
+        this->counter--;
+    }
+    if (this->counter == 0) {
+        EnIk_Thaw(this, globalCtx);
+        if (this->actor.colChkInfo.health == 0) {
+            EnIk_SetupCutscene(this);
+        } else {
+            EnIk_CheckActions(this, globalCtx);
+        }
+    }
+}
 
-extern ColliderCylinderInit D_8092C01C;
-extern ColliderTrisElementInit D_8092C048[2];
-extern ColliderTrisInit D_8092C0C0;
-extern ColliderQuadInit D_8092C0D0;
-extern DamageTable D_8092C120;
-extern DamageTable D_8092C140;
-extern CollisionCheckInfoInit D_8092C160;
-extern InitChainEntry D_8092C168[];
+void EnIk_SetupCutscene(EnIk* this) {
+    ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    this->actor.speedXZ = 0.0f;
+    if (this->actor.colChkInfo.health != 0) {
+        func_800BE504(&this->actor, &this->colliderCylinder);
+    }
+    this->actionFunc = EnIk_PlayCutscene;
+}
 
-extern UNK_TYPE D_06000CE8;
-extern UNK_TYPE D_060015F8;
-extern UNK_TYPE D_06001ABC;
-extern UNK_TYPE D_06002484;
-extern UNK_TYPE D_0600391C;
-extern UNK_TYPE D_06004A04;
-extern UNK_TYPE D_06005254;
-extern UNK_TYPE D_060057F4;
-extern UNK_TYPE D_06006294;
+void EnIk_PlayCutscene(EnIk* this, GlobalContext* globalCtx) {
+    Vec3f subCamEye;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/EnIk_Init.s")
+    this->invincibilityFrames = 12;
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene) != 0) {
+        if (this->actor.cutscene != -1) {
+            ActorCutscene_StartAndSetFlag(this->actor.cutscene, &this->actor);
+            this->subCamId = ActorCutscene_GetCurrentCamera(this->actor.cutscene);
+            subCamEye.x = (Math_SinS((this->actor.shape.rot.y - 0x2000)) * 120.0f) + this->actor.focus.pos.x;
+            subCamEye.y = this->actor.focus.pos.y + 20.0f;
+            subCamEye.z = (Math_CosS((this->actor.shape.rot.y - 0x2000)) * 120.0f) + this->actor.focus.pos.z;
+            Play_CameraSetAtEye(globalCtx, this->subCamId, &this->actor.focus.pos, &subCamEye);
+        }
+        if (this->actor.colChkInfo.health != 0) {
+            EnIk_SetupReactToAttack(this, false);
+        } else {
+            EnIk_SetupDeath(this);
+        }
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/EnIk_Destroy.s")
+void EnIk_UpdateDamage(EnIk* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s32 sp2C;
```

Comment:
```suggestion
    s32 armorBroke;
```
You might be able to move this up to the `armorBroke = 0;` line. This version of C allows us to declare new variables at the start of the inside of code blocks following if conditions, that might be what is happening here, that the variable was originally declared there not just set to zero.

Reply:
Yup, that worked! I was also able to take off one of the pads